core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -247,9 +247,15 @@ private Set<QualifiedTablePrefix> calculatePrefixesWithSchemaName(
         }
 
         Session session = ((FullConnectorSession) connectorSession).getSession();
-        return listSchemaNames(session)
+        Set<QualifiedTablePrefix> schemaPrefixes = listSchemaNames(session)
                 .filter(prefix -> predicate.get().test(schemaAsFixedValues(prefix.getSchemaName().get())))
                 .collect(toImmutableSet());
+        if (schemaPrefixes.size() > MAX_PREFIXES_COUNT) {
+            // in case of high number of prefixes it is better to populate all data and then filter
+            // TODO this may cause re-running the above filtering upon next applyFilter
+            return defaultPrefixes(catalogName);
```

Comment:
This probably fixes a bug I introduced in 677b52d8b06f4fe983e41e2fcd1e470f79930a30.
`calculatePrefixesWithSchemaName` could return many prefixes
then `calculatePrefixesWithTableName` could exit here
https://github.com/trinodb/trino/blob/14742453adbca9231449688452c9992458b4c789/core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java#L312
and then this verification would fail
https://github.com/trinodb/trino/blob/14742453adbca9231449688452c9992458b4c789/core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java#L222

Can you call the commit "Fix ..." if indeed this is a bug fix?
is it worth adding a test?

Reply:
`testSelectTablesWithLikeOverSchema` and `testSelectTablesWithLikeOverTableName` from added `TestHiveMetastoreReadAccessOperations` detect that case.

Reply:
Renamed commit message with short explanation

Reply:
@huberty89 thanks for adding

> When there is more than 100 schemas InformationSchemaMetadata#calculatePrefixesWithSchemaName should return default prefix instead of all schemas

did the code fail before?
locally see that the new test `io.trino.plugin.hive.metastore.thrift.TestHiveMetastoreReadAccessOperations#testSelectTablesWithLikeOverSchema`  would fail without this change, so i am marking this PR as a release blocker

Reply:
Yes, it was failing - exception from `verify` check.

Reply:
btw thanks for fixing my bug

Reply:
Happy to help